### PR TITLE
Add option to run a beforeMapMount hook and also save the leaflet instance on window or global

### LINKED ISSUE #35: Integration with Leaflet Geoman plugin
Hello, thanks for all your hard work. 

I have been doing some prototypes using the Vue 3 repo recently and everything was working perfect until I wanted to install https://github.com/geoman-io/leaflet-geoman . 

This plugin auto install itself on the L constant when it  it's imported. The current Vue 3 components for what I can see are dynamically importing leafleat so I haven't found a way to make the plugin be able to reference the same leaflet map object that the component is using. 

I try doing things similar to https://github.com/geoman-io/leaflet-geoman/issues/580 but given the new way that the Vue 3 library works with leaflet it's not working for me. Any help is very much appreciated! thanks

### PR DESCRIPTION
**Pull request by juan267**
Leaflet plugins community relies a lot on having the `leaflet` object available on window to initialize them selfs. Because of that is necessary to make the leaflet instance available on `window`. Also is neccesary sometimes to run code on `beforeMapMount` hook to for example configure something on the global `L` instance before creating a `map` instance. To accomplish that this PR adds a `beforeMapMount` option that will be called if available before mounting the map. 

With this changes is possible to write code like the following: 

```vue
<template>
  <l-map
    ref="map"
    v-model:zoom="zoom"
    :center="[47.41322, -1.219482]"
    :options="{ beforeMapMount }"
    @ready="mapReady"
  >
    <l-tile-layer
      url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
      layer-type="base"
      name="OpenStreetMap"
    ></l-tile-layer>
  </l-map>
</template>
<script>
import { LMap, LTileLayer } from "./../../components";

export default {
  components: {
    LMap,
    LTileLayer,
  },
  data() {
    return {
      zoom: 2,
    };
  },
  methods: {
    async mapReady(map) {
      map.pm.addControls({
        position: "topleft",
        drawCircle: false,
      });
    },
    async beforeMapMount() {
      await import("@geoman-io/leaflet-geoman-free");
      await import("@geoman-io/leaflet-geoman-free/dist/leaflet-geoman.css");
    },
  },
};
</script>

<style></style>
```

In this example the plugin https://github.com/geoman-io/leaflet-geoman it's being used. 

Fixes #35

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
-  "name": "@vue-leaflet/vue-leaflet",
```
Comment (DonNicoJs) [inline package.json:2]: I am assuming this is an error, can you remove this?

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -0,0 +1,10425 @@
+devDependencies:
```
Comment (DonNicoJs) [inline pnpm-lock.yaml:1]: Can we remove this file?

File: src/components/LMap.vue [hunk 3]
```diff
@@ -210,9 +215,17 @@ export default {
     };
 
     onMounted(async () => {
-      const { map, CRS, Icon, latLngBounds, latLng, DomEvent } = await import(
-        "leaflet/dist/leaflet-src.esm"
-      );
+      WINDOW_OR_GLOBAL.L = WINDOW_OR_GLOBAL.L || (await import("leaflet"));
+      const {
```
Comment (DonNicoJs) [inline src/components/LMap.vue:219]: Can we restore the formatting originally used? I am assuming you have format on save, in your editor

File: src/utils.js [hunk 4]
```diff
@@ -58,20 +58,18 @@ export const remapEvents = (contextAttrs) => {
...
-  ]);
-
-  delete Icon.Default.prototype._getIconUrl;
-
-  Icon.Default.mergeOptions({
-    iconRetinaUrl: modules[0].default,
-    iconUrl: modules[1].default,
-    shadowUrl: modules[2].default,
-  });
+export const resetWebpackIcon = async () => {
```
Comment (DonNicoJs) [inline src/utils.js:61]: Why removing this function and pushing commented code?

File: yarn.lock [hunk 5]
```diff
@@ -1,9794 +0,0 @@
-# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (DonNicoJs) [inline yarn.lock]: WE need this file

**Review by DonNicoJs — CHANGES_REQUESTED** [review]

**Comment by DonNicoJs** [comment]
@juan267 Thanks for taking the time to submit this work. I've left a few comments.

In addition this approach completely breaks `tree-shakability` of the app, enlarging the bundle size several time. I think we should address this concern.

**Comment by DonNicoJs** [comment]
@juan267 In addition, please be careful when creating a PR to not change more than what your intended change is, especially on submitting `package.json` changes that changes the author and the name of the library

**Comment by juan267** [comment]
>  In addition, please be careful when creating a PR to not change more than what your intended change is, especially on submitting `package.json` changes that changes the author and the name of the library

Oh I'm so sorry for this this PR got quite old and I forgot that It was based on the master branch of my fork so it pick up recent changes from there that where not intended to be here. I have recreated it with the correct changes here https://github.com/vue-leaflet/vue-leaflet/pull/90. I will close this one

**Pull request closed without merging**